### PR TITLE
Adding As and community attributes in the bgp-prefix yang model.

### PR DESCRIPTION
**Pull request by SuryyaKrJana**

### CODE REVIEW

File: models/bgp/open-traffic-generator-bgp.yang [hunk 1]
```diff
@@ -298,8 +298,121 @@ module open-traffic-generator-bgp {
...
+              config false;
+              description
+                "Opstate data for AS-PATH segments";
+              uses bgp-as-path-attr-state;
+            }
+          }
+
+          list community {
+            description
+              "Unkeyed list of Optional community attributes.";
```
Comment (apratimmukherjee) [inline models/bgp/open-traffic-generator-bgp.yang:317]: Should Optional be lower case.  Maybe add  "present in the UPDATE message".

File: models/bgp/open-traffic-generator-bgp.yang [hunk 2]
```diff
@@ -298,8 +298,121 @@ module open-traffic-generator-bgp {
...
+            container state {
+              config false;
+              description
+                "Opstate data for communities";
```
Comment (apratimmukherjee) [inline models/bgp/open-traffic-generator-bgp.yang:321]: Is Opstate the correct way to describe it in gnmi . Maybe thats the way but at first glance Opstate data did not make much sense to me.

File: models/bgp/open-traffic-generator-bgp.yang [hunk 3]
```diff
@@ -298,8 +298,121 @@ module open-traffic-generator-bgp {
...
+              description
+                "Opstate data for AS-PATH segments";
+              uses bgp-as-path-attr-state;
+            }
+          }
+
+          list community {
+            description
+              "Unkeyed list of Optional community attributes.";
+            container state {
```
Comment (apratimmukherjee) [inline models/bgp/open-traffic-generator-bgp.yang:318]: Is the container name always 'state' . Again , might be the way to do it in gnmi/openconfig but could not follow why it is this way everywhere.

File: models/bgp/open-traffic-generator-bgp.yang [hunk 4]
```diff
@@ -298,8 +298,121 @@ module open-traffic-generator-bgp {
...
+  }
+
+  grouping bgp-as-path-attr-state {
+    description
+      "AS Path segments present in the received AS Path attribute.";
+    leaf segment-type {
+      type enumeration {
+        enum AS_SEQ {
+          description
+            "Segment type is as_seq.";
```
Comment (apratimmukherjee) [inline models/bgp/open-traffic-generator-bgp.yang:337]: Should we write as as_seq or full AS Sequence . Again , not the right person to do review from gnmi/openconfig convention point of view but using full terms seems to be adding more value from documentation point view to me.

Comment (apratimmukherjee) [reply models/bgp/open-traffic-generator-bgp.yang:337]: Same comment applies to all enums here.

File: models/bgp/open-traffic-generator-bgp.yang [hunk 5]
```diff
@@ -298,8 +298,121 @@ module open-traffic-generator-bgp {
...
+      type uint32;
+      description
+        "The AS numbers in this AS path segment.";
+    }
+
+  }
+
+  grouping bgp-community-state {
+    description
+      "The bgp-community-state";
```
Comment (apratimmukherjee) [inline models/bgp/open-traffic-generator-bgp.yang:365]: The information present in one received Community attribute . Maybe ?

File: models/bgp/open-traffic-generator-bgp.yang [hunk 6]
```diff
@@ -298,8 +298,121 @@ module open-traffic-generator-bgp {
...
+    leaf community-type {
+      type enumeration {
+        enum MANUAL_AS_NUMBER {
+          description
+            "Community type is custom.";
```
Comment (apratimmukherjee) [inline models/bgp/open-traffic-generator-bgp.yang:370]: Maybe add that if Community type is 'custom' the values for the same can be retrieved by accessing the optional fields a.x and a.y ...

**Review by apratimmukherjee — APPROVED** [review]
Some doubts mentioned regarding gnmi/open-config convention and whether we are doing the right thing w.r.t. documentation at some points.

File: models/bgp/open-traffic-generator-bgp.yang [hunk 7]
```diff
@@ -298,8 +298,121 @@ module open-traffic-generator-bgp {
...
+        }
+        enum NO_LLGR {
+          description
+            "Community type is no_llgr.";
         }
       }
+      description
+        "The type of community AS number.";
+    }
+    container state {
```
Comment (apratimmukherjee) [inline models/bgp/open-traffic-generator-bgp.yang:396]: Can we directly put the custom attributes as optional field in the parent struct . Do we need another 'state' holder. Might also be causing the compilation issues.

File: models/bgp/open-traffic-generator-bgp.yang (hunk 3 above)
Comment (PrasenjitAdhikary) [reply models/bgp/open-traffic-generator-bgp.yang:318]: Agree. We should remove this additional nesting level.

**Review by PrasenjitAdhikary — APPROVED** [review]

**Pull request merged**